src/Kanbas/Courses/Quizzes/Details/index.tsx

```diff
@@ -1,35 +1,184 @@
-import { Button } from "react-bootstrap";
+import { Button, Table } from "react-bootstrap";
 import { MdOutlineEdit } from "react-icons/md";
 import MiddleContent from "../../../layout/Content/NotLeftSide/MiddleContent";
 import MiddleContentActions from "../../../layout/Content/NotLeftSide/MiddleContent/MiddleContentActions";
 import MiddleContentData from "../../../layout/Content/NotLeftSide/MiddleContent/MiddleContentData";
 import { useNavigate, useParams } from "react-router";
+import { useDispatch, useSelector } from "react-redux";
+import { KanbasState } from "../../../store";
+import { IKanbasQuizDetails } from "../../../store/interfaces/quizzes";
+import { setQuiz } from "../reducer";
+import { useEffect } from "react";
+import { getQuizDetails, postQuizSetPublish } from "../client";
+import { FaBan, FaCheckCircle, FaEllipsisV } from "react-icons/fa";
+import "./index.css";
+import { eQuizType } from "../../../store/enums/eQuizType";
 
 function QuizDetails() {
-    const { courseId } = useParams();
-    const navigate = useNavigate();
+  const { courseId, quizId } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
 
-    // TODO pass in the quiz object later and update the path
-    const handleQuizEditButton = () => {
-        navigate(`/Kanbas/Courses/${courseId}/Quizzes/quizId/edit`);
+  const quiz = useSelector((state: KanbasState) => state.quizzesReducer.quiz);
+
+  const handleEdit = async () => {
+    navigate(`/Kanbas/Courses/${courseId}/Quizzes/${quizId}/edit`);
+  };
+
+  const renderBooleanValue = (val: boolean) => {
+    if (val) return <>Yes</>;
+    else return <>No</>;
+  };
+
+  const handlePublishToggle = () => {
+    postQuizSetPublish(quiz._id, !quiz.isPublished).then(() => {
+      dispatch(setQuiz({ ...quiz, isPublished: !quiz.isPublished }));
+    });
+  };
+
+  useEffect(() => {
+    if (quizId !== undefined && courseId !== undefined) {
+      getQuizDetails(quizId).then((quiz: IKanbasQuizDetails) => {
+        console.log(quiz);
+        dispatch(setQuiz(quiz));
+      });
     }
+  }, [dispatch, quizId, courseId]);
+
+  return (
+    <>
+      <MiddleContent>
+        <MiddleContentActions>
+          <Button
+            className={
+              quiz.isPublished ? "wd-button-published" : "wd-button-standard"
+            }
+            onClick={handlePublishToggle}
+          >
+            <div className="d-flex justify-content-center align-items-center">
+              {quiz.isPublished && (
+                <>
+                  <FaCheckCircle className="me-1" /> Published
+                </>
+              )}
+              {!quiz.isPublished && (
+                <>
+                  <FaBan className="me-1" /> Publish
+                </>
+              )}
+            </div>
+          </Button>
+          <Button className="wd-button-standard me-2">
```

Comment:
Missing `onClick()` property?

Reply:
It is to be added later when the Preview component is ready.

Reply:
It could have been added now since the route is already there? Whenever that component is ready, it will just render that when we click on it. Just trying to minimize TODO future work.